pytensor/tensor/random/op.py

```diff
@@ -319,8 +319,23 @@ def make_node(self, rng, size, dtype, *dist_params):
                 "The type of rng should be an instance of either RandomGeneratorType or RandomStateType"
             )
 
+        # Fail early when size is incompatible with parameters
+        size_len = get_vector_length(size)
```

Comment:
Perhaps I should move this to `infer_shape`. If for some reason an RV needs different batching semantics it can override it?

Reply:
There was already a case with the ChoiceRV and PermutationRVs which need special handling for the shape